subsys/bluetooth/host/gatt.c

```diff
@@ -4650,7 +4650,10 @@ int bt_gatt_unsubscribe(struct bt_conn *conn,
 
 void bt_gatt_cancel(struct bt_conn *conn, void *params)
 {
-	bt_att_req_cancel(conn, params);
+	struct bt_att_req *req;
+
+	req = bt_att_find_req_by_user_data(params);
```

Comment:
There may be a race condition here. E.g. Applicaiton thread is preempted here by BT TX thread. BT TX thread sends the `req`. High priority stuff happens. BT RX thread now runs and gets the response. BT RX calls callback in application which calls `bt_gatt_write` and allocates the same `req`. Control goes back to application thread and `req` is now bad.

Maybe the "High priority stuff happens." is not something that practically happens. But this code is clearly fragile and I would not be surprised if a more likely race is found. Anyone have thoughts what on synchronization to use here?

Reply:
k_sched_lock/unlock would be acceptable here. As long as there is nothing that unschedules the thread between.
From what I can tell there is nothing that would block in bt_att_req_free.